crates/xtask/src/modify_data.rs

```diff
@@ -0,0 +1,225 @@
+        self.copy_tree(
+            &plex_config_path,
+            &plex_stub_config_path,
+            vec!["Plug-in Support", "Databases"],
+        )?;
+
+        Ok(())
+    }
+
+    fn copy_tree(
+        &self,
+        plex_config_path: &Path,
+        plex_stub_config_path: &Path,
+        paths: Vec<&str>,
+    ) -> anyhow::Result<()> {
+        let mut source = plex_config_path.to_owned();
+        let mut target = plex_stub_config_path.to_owned();
+
+        for path in paths {
+            source = source.join(path);
+            target = target.join(path);
+        }
+
+        remove_dir_all(&target)?;
+
+        for entry in WalkDir::new(&source)
+            .follow_links(false)
+            .into_iter()
+            .filter_map(|e| e.ok())
+        {
+            if entry.file_type().is_file() {
+                let path = entry.path();
+                let new_path = target.join(path.strip_prefix(&source)?);
+
+                self.cp(entry.path(), new_path)?;
+            } else if entry.file_type().is_symlink() {
+                let path = entry.path();
+                let new_path = plex_stub_config_path.join(path.strip_prefix(plex_config_path)?);
+                let link_target = path.canonicalize()?;
+
+                if link_target.strip_prefix(plex_config_path).is_ok() {
+                    self.symlink(path, &link_target, &new_path)?;
+                } else {
+                    self.cp(path, new_path)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn symlink<P: AsRef<Path>, Q: AsRef<Path>, R: AsRef<Path>>(
+        &self,
+        path: P,
+        target: Q,
+        new_path: R,
+    ) -> anyhow::Result<()> {
+        if let Some(path_parent) = path.as_ref().parent() {
+            if let Some(relative) = diff_paths(&target, path_parent) {
+                if let Some(parent) = new_path.as_ref().parent() {
+                    if !parent.exists() {
+                        create_dir_all(parent)?;
+                    }
+                }
+
+                if self.verbose {
+                    println!(
+                        "Linking '{}' to '{}'",
+                        new_path.as_ref().display(),
+                        relative.display()
+                    );
+                }
+
+                #[cfg(not(windows))]
+                {
+                    std::os::unix::fs::symlink(relative, new_path)?;
+                }
+
+                #[cfg(windows)]
+                {
+                    std::os::windows::fs::symlink_file(relative, new_path)?;
+                }
+
+                return Ok(());
+            }
+        }
+
+        eprintln!(
+            "Could not relink '{}', falling back to copying.",
+            path.as_ref().display(),
+        );
+
+        self.cp(path, new_path)
+    }
+
+    fn cp<P: AsRef<Path>, Q: AsRef<Path>>(&self, original: P, link: Q) -> anyhow::Result<()> {
```

Comment:
Same as with `symlink()`: how do you feel about moving this method to some shared place to use it inside both `modify-tags` and `plex-data`?